examples/deployment/kubernetes/deploy.sh

```diff
@@ -3,50 +3,72 @@
 # - Cluster has already been created & configured using the create.sh script
 # - Go 1.9 is installed
 
+function checkEnv() {
+  if [ -z ${PROJECT_NAME+x} ] ||
+     [ -z ${CLUSTER_NAME+x} ] ||
+     [ -z ${MASTER_ZONE+x} ] ||
+     [ -z ${CONFIG_MAP+x} ]; then
+    echo "You must either pass an argument which is a config file, or set all the required environment variables"
+    exit 1
+  fi
+}
+
 set -e
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-source ${DIR}/config.sh
+if [ $# -eq 1 ]; then
+  source $1
+else
+  checkEnv
+fi
+
 
 export LOG_URL=TODO
 export MAP_URL=TODO
-export TAG=$(git rev-parse HEAD)
+export IMAGE_TAG=${IMAGE_TAG:-$(git rev-parse HEAD)}
+
+# Connect to gcloud
+gcloud --quiet config set project ${PROJECT_NAME}
+gcloud --quiet config set container/cluster ${CLUSTER_NAME}
+gcloud --quiet config set compute/zone ${MASTER_ZONE}
+gcloud --quiet container clusters get-credentials ${CLUSTER_NAME}
+
+# Configure Docker to use gcloud credentials with Google Container Registry
+gcloud auth configure-docker
 
 # Get Trillian
 go get github.com/google/trillian/...
 cd $GOPATH/src/github.com/google/trillian
 
-# Build docker images
-docker build -f examples/deployment/docker/log_server/Dockerfile -t gcr.io/$PROJECT_NAME/log_server:$TAG .
-docker build -f examples/deployment/docker/log_signer/Dockerfile -t gcr.io/$PROJECT_NAME/log_signer:$TAG .
+echo "Building docker images..."
+docker build --quiet -f examples/deployment/docker/log_server/Dockerfile -t gcr.io/$PROJECT_NAME/log_server:$IMAGE_TAG .
+docker build --quiet -f examples/deployment/docker/log_signer/Dockerfile -t gcr.io/$PROJECT_NAME/log_signer:$IMAGE_TAG .
 # TODO(al): when cloudspanner supports maps:
 # docker build -f examples/deployment/docker/map_server/Dockerfile -t us.gcr.io/$PROJECT_NAME/map_server:$TAG .
 
-# Connect to gcloud
-gcloud config set project "${PROJECT_NAME}"
-gcloud config set compute/zone "${ZONE}"
-
-# Configure Docker to use gcloud credentials with Google Container Registry
-gcloud auth configure-docker
+echo "Pushing docker images..."
+gcloud docker -- push gcr.io/${PROJECT_NAME}/log_server:${IMAGE_TAG}
+gcloud docker -- push gcr.io/${PROJECT_NAME}/log_signer:${IMAGE_TAG}
 
-# Push docker images
-docker push "gcr.io/${PROJECT_NAME}/log_server:${TAG}"
-docker push "gcr.io/${PROJECT_NAME}/log_signer:${TAG}"
+echo "Tagging docker images..."
+gcloud --quiet container images add-tag gcr.io/${PROJECT_NAME}/log_server:${IMAGE_TAG} gcr.io/${PROJECT_NAME}/log_server:latest
+gcloud --quiet container images add-tag gcr.io/${PROJECT_NAME}/log_signer:${IMAGE_TAG} gcr.io/${PROJECT_NAME}/log_signer:latest
```

Comment:
Why do they need to be explicitly tagged as "latest"?

Reply:
That's currently what the `*-deployment.yaml` files look for, is there a better way?

Reply:
Further down, you use `kubectl set image` to override the default image (i.e. the image with the "latest" tag) and use `${IMAGE_TAG}` instead. Actually, it might be best to replace "latest" in `*-deployment.yaml` with an env var that `envsubst` can substitute. That could be done as a separate PR though.